TypeCobol/Compiler/Parser/CodeElementBuilder/CobolExpressionsBuilder.cs

```diff
@@ -703,15 +705,16 @@ internal ConditionalExpression CreateClassCondition(CodeElementsParser.ClassCond
 					ParseTreeUtils.GetFirstToken(context.dataItemContentType()));
 
 				classCondition = new ClassCondition(
-					CreateIdentifier(context.identifier()),
+					new ConditionOperand(new Variable(CreateIdentifier(context.identifier()))),
 					dataItemContentType,
 					invertResult);
 			}
 
             // Collect storage area read/writes at the code element level
             if (classCondition.DataItem != null)
             {
-                this.storageAreaReads.Add(classCondition.DataItem);
+                Debug.Assert(classCondition.DataItem.Variable?.StorageArea != null); // This is how we created it above
```

Comment:
It's not a new `Debug.Assert` but instead the condition must be updated. In fact you should factorize the storage area creation outside of the main `if`. From there you'll have a variable to test in order to update the dictionary. Easier.